docs/providers/datadog.md

```diff
@@ -0,0 +1,155 @@
+This method makes it more flexible to input any `Datadog` SLI computation and
+eventually reduces the number of queries made to Datadog.
+
+```yaml
+backend:
+  class:   Datadog
+  method:  query_sli
+  api_key: ${DATADOG_API_KEY}
+  app_key: ${DATADOG_APP_KEY}
+  measurement:
+    expression: sum:system.disk.used{*} / sum:system.disk.total{*}
+```
+
+**&rightarrow; [Full SLO config](../samples/datadog/slo_dd_disk_utilization_query_sli.yaml)**
+
+### Query SLO
+
+The `query_slo` method is used to directly query the needed SLO with Datadog:
+indeed, Datadog has SLO objects that you can directly refer to in your config by inputing their `slo_id`.
+
+This method makes it more flexible to input any `Datadog` SLI computation and
+eventually reduces the number of queries made to Datadog.
+
+To query the value from Datadog SLO, simply add a `slo_id` field at the top level of your configuration.
+
+```yaml
+slo_id:  ${DATADOG_SLO_ID}
+...
+backend:
+  class:   Datadog
+  method:  query_sli
+  api_key: ${DATADOG_API_KEY}
+  app_key: ${DATADOG_APP_KEY}
+```
+
+**&rightarrow; [Full SLO config](../samples/datadog/slo_dd_disk_utilization_query_slo.yaml)**
+
+### Examples
+
+Complete SLO samples using `Datadog` are available in
+[samples/datadog](../samples/datadog). Check them out !
+
+## Exporter
+
+The `Datadog` exporter allows to export the error budget burn rate metric as
+a **Datadog metric** that can be used for alerting:
+
+ * The **metric name** is `error_budget_burn_rate` by default, but can be
+ modified using the `metric_type` field in the exporter YAML.
+
+ * The **metric descriptor** has labels describing our SLO, amongst which the
+ `service_name`, `feature_name`, and `error_budget_policy_step_name` labels.
+
+The exporter pushes the metric to the `Datadog` API.
+
+**Example config:**
+
+```yaml
+exporters:
+ - class: Datadog
+   api_key: ${DATADOG_API_KEY}
+   app_key: ${DATADOG_APP_KEY}
+```
+
+Optional fields:
+  * `metric_type`: Metric type / name. Defaults to `error_budget_burn_rate`.
+  * `metric_description`: Metric description.
+
+**&rightarrow; [Full SLO config](../samples/datadog/slo_dd_disk_utilization_ratio.yaml)**
+
+
+## Datadog API considerations
+
+The `distribution_cut` method is not currently implemented for Datadog.
+
+The reason for this is that Datadog distributions (or histograms) do not conform
+to what histograms should be (see [old issue](https://github.com/DataDog/dd-agent/issues/349)),
+i.e a set of configurable bins, each providing the number of events falling into
+each bin.
+
+Standard histograms representations (see [wikipedia](https://en.wikipedia.org/wiki/Histogram))
+already implement this, but the approach Datadog took is to pre-compute
+(client-side) or post-compute (server-side) percentiles, resulting in a
+different metric for each percentile representing the percentile value instead
+of the number of events in the percentile.
+
+This implementation has a couple advantages, like making it easy to query and
```

Comment:
"has a couple advantages" -> "has a couple of advantages" or just "has advantages"

Reply:
Fixed